src/kvtree_mpi_io.c

```diff
@@ -133,6 +133,19 @@ static unsigned long kvtree_write_gather_map(
 
   /* get name of this process */
   int rank_world, ranks_world;
+
+  if (offset != 0) {
+    /*
+     * While we technically can write to a non-zero offset, there were issues
+     * with doing so with NFS, and so we disable it for now.  It will break
+     * kvtree_read_scatter_single() if we enable it too.
+     */
+    kvtree_err("Attempt to write a non-zero offset (got %lu) for %s/%s @ %s:%d",
+      offset, prefix, file, __FILE__, __LINE__
+    );
+    return KVTREE_FAILURE;
+  }
+
   MPI_Comm_rank(comm, &rank_world);
   MPI_Comm_size(comm, &ranks_world);
```

Comment:
Good idea to add this check.  Let's move it in front of the "get rank/ranks" block rather than splitting it in half.
```suggestion
  if (offset != 0) {
    /*
     * While we technically can write to a non-zero offset, there were issues
     * with doing so with NFS, and so we disable it for now.  It will break
     * kvtree_read_scatter_single() if we enable it too.
     */
    kvtree_err("Attempt to write a non-zero offset (got %lu) for %s/%s @ %s:%d",
      offset, prefix, file, __FILE__, __LINE__
    );
    return KVTREE_FAILURE;
  }

  /* get name of this process */
  int rank_world, ranks_world;
  MPI_Comm_rank(comm, &rank_world);
  MPI_Comm_size(comm, &ranks_world);
```

Reply:
Oh, let's also call ``kvtree_abort`` here instead of ``kvtree_err``.  In the case where offset values are used, different processes get different values, but usually at least one process has offset=0.  In that case, we'd end up with a deadlock if only a subset of procs return early.  It's better to abort.